frontend/src/app/components/wp-fast-table/state/wp-table-columns.service.ts

```diff
@@ -87,10 +87,11 @@ export class WorkPackageTableColumnsService extends WorkPackageTableBaseService<
   }
 
   /**
-   * Retrieve the QueryColumn objects for the selected columns
+   * Retrieve the QueryColumn objects for the selected columns.
+   * Returns a shallow copy with the original column objects.
    */
   public getColumns():any[] {
-    return (this.currentState && this.currentState.getColumns()) || [];
+    return [ ...this.currentState.getColumns() ];
```

Comment:
Why is this the same? Where is the check, that `this.currentState` is available?

Reply:
This returns a [shallow copy using the spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). You're right however that this may be undefined initially, so we could instead return `[]` in `currentState`